### PR TITLE
Add Cockroach DB docker compose and examples

### PR DESCRIPTION
**Pull request by samredai**
### Summary

This adds an alternative docker compose setup that uses Cockroach DB for both the DJ metadata as well as the examples. Using overrides, it disables the postgres containers in the default `docker-compose.yml` file and replaces them with cockroach db containers. (also included a section to the examples/README.rst that describes these overrides).
```
docker compose -f docker-compose.yml -f docker-compose.cockroachdb.yml up
```
The Cockroach DB UI's are made available at http://localhost:8080 (metadata) and http://localhost:8081 (examples).
These are great for exploring generated tables and SQL activity.

The examples are based on a sample of some steam data--hours played by users, game information, etc.

### Test Plan

I ran the docker compose command with the cockroachdb override and queried for various metrics through the `/graphql` server. One issue that can be fixed separately from this PR is the way we generate the database and node IDs. That seemed to create issues caused by Cockroach DB generating very large integers that I describe in a [comment](https://github.com/DataJunction/dj/issues/193#issuecomment-1275569884) to Issue #193.

- [x] PR has an associated issue: #218
- [x] `make check` passes
- [x] `make test` shows 100% unit test coverage

### Deployment Plan

No special deployment instructions.

### CODE REVIEW

**Comment by samredai** [comment]
cc: @CircArgs @betodealmeida @agorajek

**Review by betodealmeida — APPROVED** [review]
So cool! Love that this opens up for a lot of different setups in the future!

File: Makefile [hunk 1]
```diff
@@ -20,6 +20,9 @@ docker-run:
 docker-run-with-druid:
 	docker compose -f docker-compose.yml -f docker-compose-druid.yml up
 
+docker-run-with-cockroachdb:
+	docker compose -f docker-compose.yml -f docker-compose.cockroachdb.yml up
```
Comment (agorajek) [inline Makefile:24]: nit: either make it `docker-compose-cockroachdb.yml` or `docker-compose.druid.yml`

File: examples/README.rst [hunk 2]
```diff
@@ -96,6 +96,22 @@ And you should see:
...
+in Postgres. However, DJ leverages SQL Alchemy to enable flexibility when it comes to reading source databases as well as storing
+its own metadata. Variations of the default docker compose environments can be selected using one of the available override
+docker compose files which add, remove, or modify services.
+
++-------------------+----------------------------------------------------------------------------+------------------------------------------------------------------------+
+| Name              | Command                                                                    | Description                                                            |
++===================+============================================================================+========================================================================+
+| Postgres          | docker compose up                                                          | DJ server backed by Postgres                                           |
+| Postgres + Druid  | docker compose -f docker-compose.yml -f docker-compose-druid.yml up        | An extension of the Postgres setup that includes Druid                 |
```
Comment (agorajek) [inline examples/README.rst:111]: ditto

**Review by agorajek — APPROVED** [review]
Love it! Thanks @samredai . Just one nit on the docker file naming.

File: Makefile (hunk 1 above)
Comment (samredai) [reply Makefile:24]: Fixed, good catch 👍

File: examples/README.rst (hunk 2 above)
Comment (samredai) [reply examples/README.rst]: Fixed

**Review by CircArgs — APPROVED** [review]

**Pull request merged**